src/prognostic_variables.jl

```diff
@@ -96,3 +96,85 @@ function Base.zeros(::Type{PrognosticVariables{NF}},
     pres = zeros(SurfaceLeapfrog{NF},_lmax,_mmax)                     # 2 leapfrog time steps for pres
     return PrognosticVariables(layers,pres,lmax,mmax,N_LEAPFROG,nlev)
 end
+
+# SET_VAR FUNCTIONS TO ASSIGN NEW VALUES TO PrognosticVariables
+
+function set_var!(progn::PrognosticVariables{NF}, 
+                  varname::Symbol, 
+                  var::Vector{<:LowerTriangularMatrix},
+                  lf::Integer=1) where NF
+
+    @assert length(var) == length(progn.layers)
+
+    for (progn_layer, var_layer) in zip(progn.layers, var)
+        _set_var_core!(getfield(progn_layer.leapfrog[lf], varname), var_layer)
+    end 
+
+    return progn 
+end 
+
+function _set_var_core!(var_old::LowerTriangularMatrix{T}, var_new::LowerTriangularMatrix{R}) where {T,R}
```

Comment:
Should be also allow `Matrix{R}` here? In case someone doesn't bother doing `LowerTriangularMatrix(::Matrix)`?

Reply:
That's what the other functions are for, they handle grid input. Or do you mean `Matrix` to input SPHs? That could get a bit messy because in other parts we also assume that Matrices are in grid space

Reply:
Yes, true. No, users can do `LowerTriangularMatrix(::Matrix)` manually then for the spherical harmonic coefficients